model/roles.go

```diff
@@ -696,51 +707,94 @@ func (r *Role) GetRoleDevVersion(opinions *Opinions, tagExtra, fissileVersion st
 		sort.Strings(keys)
 
 		// ... then add them and their values to the hash precursor
+		// For the graph output, adding all properties individually results in
+		// too many nodes and makes graphviz fall over. So use the hash of them
+		// all instead.
+		propertyHasher := sha1.New()
```

Comment:
Should this be conditional on `grapher != nil` ?

Reply:
I skipped this one because it's not doing much work, and adding the check complicates the flow a bit (for reading).

Reply:
Ok